### ISSUE TITLE
Feature Request: WRITE CSV and WRITE JSON

### ISSUE THREAD

**Issue by jmatsushita** [neo4j/neo4j#2622]
Hi there,

With this great ETL feature you've developed for 2.1 how about also allowing to export data using the flexibility of Cypher? Could look like

```
WRITE CSV WITH HEADERS 
TO 'file:///path/to/nodes.csv'
MATCH (n)
RETURN n.id as id, n.label as label, n.description as description
TO 'file:///path/to/edges.csv'
MATCH (s)-[r]->(t)
RETURN r.id as id, s.label as source, t.label as target
```

which would create a `nodes.csv`

```
id, name, description
n1,n1, "blah blah"
...
```

and a `edges.csv`

```
id, source, target
1,n1,n2
...
```

Or 

```
WRITE JSON 
TO 'file:///path/to/export.json#nodes'
MATCH (n)
RETURN n.id as id, n.label as label, n.description as description
TO 'file:///path/to/export.json#edges'
MATCH (s)-[r]->(t)
RETURN r.id as id, s.label as source, t.label as target
```

Which would create

```
{
    "nodes": [
        {
            "id": "n1",
            "label": "n1",
            "description": "blah blah"
        },
        ...
        ]
     "edges": [
        {
           "id": "1",
           "source": "n1",
           "target": "n2"
        },
        ...
     ]
}
```

You could return the raw json directly from a REST call instead of writing to file with a syntax like this:

```
WRITE JSON 
TO '#nodes'
MATCH (n)
RETURN n
TO '#edges'
MATCH ()-[r]->()
RETURN r
```

It should then be easy to create GraphSON or other exotic representations of graphs in JSON, CSV (and... I don't care so much for it but I guess you could throw XML in there for GraphML or GEXF...).

Basically https://github.com/jexp/neo4j-shell-tools on steroids. 

Cheers,

Jun

Related to structr/structr/issues/186
